charts/linkerd2/templates/prometheus.yaml

```diff
@@ -118,6 +118,10 @@ data:
       # foo=bar
       - action: labelmap
         regex: __meta_kubernetes_pod_label_linkerd_io_(.+)
+      - action: labelmap
```

Comment:
It took me a few minutes to wrap my head around these rules but I think I get it.

First we add all pod labels that start with `linkerd_io`, dropping the `linkerd_io` prefix.  So for example, the `linkerd.io/foobar` pod label becomes `foobar` as a Prometheus label.

Next we add all other pod labels.  So the pod label `bas` becomes `bas` as a Prometheus label.  Unfortunately, this also copies those `linkerd.io` labels as well so the `linkerd.io/foobar` pod label from above also becomes `linkerd_io_foobar` as a Prometheus labels.

To correct this, we now drop all labels which start with `linkerd_io_`.  

This works pretty well but it is a bit confusing and may lead to a "gotcha" in the future:

If, in the future, we have any Prometheus labels (not necessarily from pod labels) which start with `linkerd_io_`, they will get dropped.

What do you think about changing this to something like

```
      # Copy all pod labels to tmp labels
      - action: labelmap
        regex: __meta_kubernetes_pod_label_(.+)
        replacement: __tmp_pod_label_$1
      # Drop `linkerd_io_` prefix from any of those tmp labels
      - action: labelmap
        regex: __tmp_pod_label_linked_io_(.+)
        replacement:  __tmp_pod_label_$1
      # Copy tmp labels into real labels
      - action: labelmap
        regex: __tmp_pod_label_(.+)
```

Do you think that makes it clearer what is happening?

Reply:
@adleong Note this will also pick the stock labels like `linkerd.io/control-plane-ns: linkerd`, leaving us with duped labels: their complete version and their stripped version (stripped in the "foo=bar" rule above).

Reply:
@alpeb you're right, I forgot that the second action there won't delete the old label that includes the prefix.  Let me experiment a bit and see if I can come up with something that works...

Reply:
Ok, this is what I've come up with:

```
      # Copy all pod labels to tmp labels
      - action: labelmap
        regex: __meta_kubernetes_pod_label_(.+)
        replacement: __tmp_pod_label_$1
      # Take `linkerd_io_` prefixed labels and copy them without the prefix
      - action: labelmap
        regex: __tmp_pod_label_linked_io_(.+)
        replacement:  __tmp_pod_label_$1
      # Drop the `linkerd_io_` originals
      - action: labeldrop
        regex: __tmp_pod_label_linkerd_io_(.+)
      # Copy tmp labels into real labels
      - action: labelmap
        regex: __tmp_pod_label_(.+)
```

This behaves almost identically to what you have, @daxmc99.  The only difference is that it won't drop any existing prometheus labels that start with `linkerd_io_` because all the dropping happens on the tmp labels.  But I think what's most important is clarity.  @daxmc99 and @alpeb, do either of you have an opinion on which of these options is easier to understand?

Reply:
I like this approach much better. A quick smoke test and this achieves the same result I was looking for.

Reply:
I'm still seeing labels with the `linkerd_io_` prefix.  I think you need to add a step to drop these from the tmp labels.  See the action labeled `# Drop the `linkerd_io_` originals` in my comment above.